compiler/rustc_codegen_cranelift/src/intrinsics/simd.rs

```diff
@@ -1008,6 +1009,56 @@ pub(super) fn codegen_simd_intrinsic_call<'tcx>(
             }
         }
 
+        sym::simd_masked_load => {
+            intrinsic_args!(fx, args => (val, ptr, mask); intrinsic);
+
+            let (val_lane_count, val_lane_ty) = val.layout().ty.simd_size_and_type(fx.tcx);
+            // let (ptr, ptr_value) = ptr.try_to_ptr().expect("simd_masked_load ptr");
+            let (mask_lane_count, _mask_lane_ty) = mask.layout().ty.simd_size_and_type(fx.tcx);
+            let (ret_lane_count, ret_lane_ty) = ret.layout().ty.simd_size_and_type(fx.tcx);
+            assert_eq!(val_lane_count, mask_lane_count);
+            assert_eq!(val_lane_count, ret_lane_count);
+
+            let lane_clif_ty = fx.clif_type(val_lane_ty).unwrap();
+            let ret_lane_layout = fx.layout_of(ret_lane_ty);
+            let ptr_val = ptr.load_scalar(fx);
+
+            for lane_idx in 0..ret_lane_count {
+                let val_lane = val.value_lane(fx, lane_idx).load_scalar(fx);
+                let mask_lane = mask.value_lane(fx, lane_idx).load_scalar(fx);
+
+                let if_enabled = fx.bcx.create_block();
+                let if_disabled = fx.bcx.create_block();
+                let next = fx.bcx.create_block();
+                let res_lane = fx.bcx.append_block_param(next, lane_clif_ty);
+
+                fx.bcx.ins().brif(mask_lane, if_enabled, &[], if_disabled, &[]);
+                fx.bcx.seal_block(if_enabled);
+                fx.bcx.seal_block(if_disabled);
+
+                fx.bcx.switch_to_block(if_enabled);
+                let offset = lane_idx as i32 * lane_clif_ty.bytes() as i32;
```

Comment:
@bjorn3 Could you have a look please? I had no idea what I was doing here. I changed `ptr` expect a scalar value and then offset it by an immediate value for each lane. How can I run run-pass tests in `tests/ui` under cranelift, to see if it works?

Reply:
I think what you wrote will work.

> How can I run run-pass tests in tests/ui under cranelift, to see if it works?

I think you can set `codegen-backend` in `config.toml` to `["cranelift"]` and then run `./x.py test tests/ui`. A fair amount of tests are expected to fail (see https://github.com/rust-lang/rust/blob/master/compiler/rustc_codegen_cranelift/scripts/test_rustc_tests.sh for a list of them), but for as long as the new tests pass, it should be fine.

Reply:
It is fine to leave it. It can't do any harm until portable-simd starts using it and when I sync the changes to the rust-lang/rustc_codegen_cranelift repo, I run the rustc test suite myself. I can then fix it if it turns out to be broken.

Reply:
The test passed when I removed the `simd_masked_store` usage from it. Currently implementing that intrinsic.

Reply:
Thanks for the update. Let me know if you need anything. The implementation of `simd_masked_store` should look a lot like `simd_scatter`, but with lanes all offset from the same pointer as opposed to having a pointer for each lane.

Reply:
Just finished and pushed it :) But thanks for the offer.